Support lidarr

gbagnoli opened the pull request:
This adds support for lidarr. It has been working for me well in the past few days.

main difference is that the API endpoint for history is slightly different (URL and returned json) and add support for AUDIO file type.

is_imported might get a slight improvement if to add the service only for audios, instead of checking all services (lidarr, sonarr, radarr, whisparr) for each file. But I tried to make the least amount of changes.

gbagnoli committed:
Support lidarr

wouterdebie commented:
Thanks for this! Rather than making an exception for lidarr you could pass in the application and make a decision based on that? It kind of looks strange to have a Boolean in a function signature to change it's behavior.

gbagnoli committed:
Refactor arr apps code

JustSuperHuman commented:
Love seeing this project get worked on! 💪

gbagnoli commented:
@wouterdebie you are right, it wasn't great.

I took a stab into moving the code in the services::arr file into a struct, so now we have an enum for the type of app (lidarr, sonarr, etc) and can have different behaviour based on that. Not sure if I went too far - let me know in case

I still think we can avoid checking lidarr for video files and the other three (sonarr, radarr, whisparr) if audio, but I can open another pull for that.

gbagnoli committed:
Allow arr apps to check for specific media types

Essentially, audio files are handled by lidarr, but not by the rest of
the supporter *arr (Sonarr, Radarr, Whisparr)
So this adds a MediaType enum for app to skip checking file types
not handled by them

gbagnoli committed:
run cargo fmt

wouterdebie commented:
Thanks @gbagnoli — landed Lidarr support on main in 1734839 (need to adapt to the [arrs.<name>] extension that landed in dd7c02e, so I ported the structural ideas — ArrKind, MediaType-tagged DownloadTargets, Lidarr's v1/trackFileImported, AUDIO handling — and credited you in the commit). Closing in favor of that commit.
